wireless_modules/mqtt_test/main.py

```diff
@@ -0,0 +1,108 @@
+import time
+import machine
+import config
+import ubinascii
+import fake
+from machine import Pin
+
+# Use micro pip to install packages
+import upip
+upip.install("micropython-umqtt.simple2")
+from umqtt.simple import MQTTClient
+
+
+# Define module number and MQTT topics
+MODULE_NUM = "1"
+FAKE_MODULE = fake.module(MODULE_NUM)
+
+PUB_DATA = b'/v3/wireless-module/{}/data'.format(MODULE_NUM)
+PUB_BATTERY = b'/v3/wireless-module/{}/battery'.format(MODULE_NUM)
+PUB_LOW_BATTERY = b'/v3/wireless-module/{}/low-battery'.format(MODULE_NUM)
+
+SUB_LED = b'/v3/wireless-module/{}/LED'.format(MODULE_NUM)
+SUB_TOPICS = [SUB_LED]
+
+# Define built in LED
+builtin_LED = Pin(2, Pin.OUT)
+
+# Set up MQTT client by generating a client_id
+client_id = ubinascii.hexlify(machine.unique_id())
+
+
+# MQTT callbacks
+def sub_cb(topic, msg):
+    print((topic, msg))
+
+    if topic == SUB_LED and msg == b'on':
+        print('ESP received: LED ON')
+        builtin_LED.on()
+
+    if topic == SUB_LED and msg == b'off':
+        print('ESP received: LED OFF')
+        builtin_LED.off()
+
+def connect_and_subscribe(client_id, mqtt_server, sub_topics):
+    client = MQTTClient(client_id, mqtt_server)
+    client.set_callback(sub_cb)
+    client.connect()
```

Comment:
What happens if connection fails?

Reply:
It tries again. Ill comment it to make it easier to read